packages/dashboard/src/pages/Wallets/components/HistoryTable/index.tsx

```diff
@@ -0,0 +1,161 @@
+            isLoading={transactionLoading}
+            isEmpty={!!transactionError || !transactions.length}
+            dataSource={transactions}
+        />
+    )
+})
+
+export interface HistoryTableUIProps {
+    page: number
+    onPageChange: Dispatch<SetStateAction<number>>
+    hasNextPage: boolean
+    isLoading: boolean
+    isEmpty: boolean
+    dataSource: Transaction[]
+}
+
+export const HistoryTableUI = memo<HistoryTableUIProps>(
+    ({ isLoading, isEmpty, dataSource, page, onPageChange, hasNextPage }) => {
+        const t = useDashboardI18N()
+
+        const classes = useStyles()
+
+        return (
+            <>
+                <TableContainer className={classes.container}>
+                    {isLoading || isEmpty ? (
+                        <Box flex={1}>
+                            {isLoading ? <LoadingPlaceholder /> : null}
+                            {isEmpty ? <EmptyPlaceholder children={t.wallets_empty_history_tips()} /> : null}
+                        </Box>
+                    ) : (
+                        <Table stickyHeader sx={{ padding: '0 44px' }}>
+                            <TableHead>
+                                <TableRow>
+                                    <TableCell key="Types" align="center" variant="head" className={classes.header}>
+                                        {t.wallets_history_types()}
+                                    </TableCell>
+                                    <TableCell key="Value" align="center" variant="head" className={classes.header}>
+                                        {t.wallets_history_value()}
+                                    </TableCell>
+                                    <TableCell key="Time" align="center" variant="head" className={classes.header}>
+                                        {t.wallets_history_time()}
+                                    </TableCell>
+                                    <TableCell key="Receiver" align="center" variant="head" className={classes.header}>
+                                        {t.wallets_history_receiver()}
+                                    </TableCell>
+                                </TableRow>
+                            </TableHead>
+
+                            {dataSource.length ? (
+                                <TableBody>
+                                    {dataSource.map((transaction, index) => (
+                                        <HistoryTableRow key={index} transaction={transaction} />
+                                    ))}
+                                </TableBody>
+                            ) : null}
+                        </Table>
+                    )}
+                </TableContainer>
+
+                {!(page === 0 && dataSource.length === 0) && !isLoading ? (
+                    <TablePagination
+                        count={-1}
+                        component="div"
+                        onPageChange={() => {}}
```

Comment:
no response on page change?

Reply:
Only the back button and next button are actually used here, so the corresponding events are in `backIconButtonProps` and `nextIconButtonProps`

Reply:
plz use `lodash.noop`